src/Nethermind/Nethermind.Trie/TrieNode.Visitor.cs

```diff
@@ -68,6 +66,25 @@ internal void Accept(ITreeVisitor visitor, ITrieNodeResolver nodeResolver, TrieV
                         }
                     }
 
+                    visitor.VisitBranch(this, trieVisitContext);
+                    trieVisitContext.Level++;
+                    if (trieVisitContext.Level <= visitor.ParallelLevels)
```

Comment:
Visitor should not be the one deciding on how to visit. Pattern seems broken.

Reply:
Disagree - visitor knows if its threadsafe and allows parallel non-ordered excecution, or its not thread-safe and allows only sequential visiting.